worlds/kdl3/Regions.py

```diff
@@ -39,22 +39,24 @@
 }
 
 
-def generate_valid_level(world: "KDL3World", level, stage, possible_stages, placed_stages):
+def generate_valid_level(world: "KDL3World", level: int, stage: int,
+                         possible_stages: List[int], placed_stages: List[int]):
     new_stage = world.random.choice(possible_stages)
     if level == 1:
         if stage == 0 and new_stage in first_stage_blacklist:
             return generate_valid_level(world, level, stage, possible_stages, placed_stages)
-        elif not (world.multiworld.players > 1 or world.options.consumables or world.options.starsanity) and \
-                new_stage in first_world_limit and \
-                sum(p_stage in first_world_limit for p_stage in placed_stages) >= 2:
+        elif (not (world.multiworld.players > 1 or world.options.consumables or world.options.starsanity) and
+                new_stage in first_world_limit and
+                sum(p_stage in first_world_limit for p_stage in placed_stages)
+              >= (2 if world.options.open_world else 1)):
```

Comment:
This condition is already huge and hard to read, I'd put at least the `2 if world.options.open_world else 1` in a local variable "max_first_world_stages" (or however you'd prefer to call it)